R/commonTTest.R

```diff
@@ -443,3 +444,108 @@ summarySEwithin <- function(data=NULL, measurevar, betweenvars=NULL, withinvars=
   # Combine the un-normed means with the normed results
   merge(datac, ndatac)
 }
+
+.descriptivesPlotsRainCloudFill <- function(dataset, variable, groups = NULL, yLabel, dependency = c("none", "paired", "repeated"), horiz) {
```

Comment:
For the paired t-test, the data comes in a wide format, whereas for the RM Anova, it comes in a long format. Because of that, I need to treat these cases separately. Do you maybe have an idea how to do that more efficiently?
I added \texttt{dependency <- match.arg(dependency)}

Reply:
Ok then maybe it would make sense instead to have a boolean option `addLines` that indicates whether to draw the lines for the repeated designs, and another argument `dataFormat = c("long", "wide")`. I think that would make it clearer on what is actually happening.

Or, you could just assume that the data come in a long format and just reshape the data in the constructor function.

Reply:
Alright, I moved the reformatting into the constructor functions for the paired t-tests and changed the arguments to `addLines` (an `xLabel` argument was also necessary)!